server/handler/frontend.go

```diff
@@ -100,3 +106,11 @@ func getRequires(result *common.Result, githubURL string) map[string][]Membershi
 	}
 	return membershipInfo
 }
+
+func getPermissions(result *common.Result) []string {
+	permStrings := make([]string, len(result.Requires.Permissions))
+	for _, perm := range result.Requires.GetPermissions() {
+		permStrings = append(permStrings, perm.String())
+	}
```

Comment:
When allocating a slice upfront and then filling it, you have two options

1. Use `make([]string, 0, length)`, then use `append` in the loop
2. Use `make([]string, length)`, then set elements by index in the loop

As written, this will create a slice with `len(result.Requires.Permissions)` empty strings, then append the new string on to the end of that.

I think you'll also want to initialize the slice using the length of `result.Requires.GetPermissions()` instead of `result.Requires.Permissions`.

All that said, another option is to remove this function and turn the `pull.Permission` types in to strings in the template, using the `printf` template function.

Reply:
Fixed, using option 1.